src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -3879,14 +3844,25 @@ namespace InternalHeatGains {
                         ShowContinueError("For " + cAlphaFieldNames(3) + "= FlowControlWithApproachTemperatures, " + cAlphaFieldNames(14) +
                                           " is required, but this field is blank.");
                         ErrorsFound = true;
-                    } else {
-                        ZoneITEq(Loop).SupplyAirNodeNum = 0;
                     }
                 } else {
                     ZoneITEq(Loop).SupplyAirNodeNum = GetOnlySingleNode(
                         AlphaName(14), ErrorsFound, CurrentModuleObject, AlphaName(1), NodeType_Air, NodeConnectionType_Sensor, 1, ObjectIsNotParent);
                 }
 
+                // Post-assign-SupplyAirNode-check. may be redundant
```

Comment:
I need a second set of eyes here. This is only needed if there's a chance ```GetOnlySingleNode``` returns a 0. If it never returns 0, then this should not be needed.

Reply:
If Alpha(14) is not blank then GetOnlySingleNode won't return 0. It will either return the node number of a real supply air node in a zone, or a non-0 number representing the Alpha(14) string (which could be mis-spelled). What is more important is that the supply air node in the ITE object match a real zone supply node.

From 1ZoneDataCenterCRAC_wApproachTemp.idf:

    ElectricEquipment:ITE:AirCooled,
      Main Zone Inlet Node,    !- Supply Air Node Name

    ZoneHVAC:EquipmentConnections,
      Main Zone Inlet Node,    !- Zone Air Inlet Node or NodeList Name

Reply:
Maybe I have that backwards. If the node name is mis-spelled then it would return a 0. I'd have to test that to be sure.

Reply:
Thanks, @rraustad. I'll do some testing on that to confirm behavior.

Reply:
If it has to match a zone supply node, then shouldn't ```NodeConnectionType_ZoneInlet``` be the correct connection type in ```GetOnlySingleNode```?

Reply:
No, there has to be an inlet and an outlet for the same node name.

Reply:
I see, so they used `NodeConnectionType_Sensor` to avoid conflicts with the other inlet and outlet nodes using this same node name. The zone inlet node and the TU outlet node, and then this node used here which is the 3rd occurrence of this node name.

Reply:
I guess there's also ```ZodeConnectionType_ZoneNode```, which is what ```GetZoneEquipmentData``` is using to set the ```ZoneNode``` variable. I'm not sure if that would be better, or if it even matters for this.

It looks like we should be checking node number returned by ```GetOnlySingleNode``` against ```GetControlledZoneIndex```. If they don't match and the Air Node Connection Type == "AdjustedSupply" or the Calculation Method == "FlowControlWithApproachTemperatures", fatal. Otherwise, pass by, or maybe even throw a warning to say that the non-blank name given doesn't match any zone node names.

Reply:
Yes, but it's node number returned against the zone's `supply` air nodes.

Reply:
@mitchute @rraustad `NodeConnectionType_Sensor` is correct here, because this object is only getting information from that node (if I remember correctly). Don't get hung up on the node connect types here, to confirm this is really a zone inlet node, query the pertinent `ZoneEquipConfig(ZoneNum).InletNode` (which is an array of inlet node nums).

Reply:
Thanks, @mjwitte. I did find the ```ZoneEquipConfig``` data and I think I've gotten the logic down. I'm working on unit tests now and hoping to push soon.